File: GXpert/GXpert.Web/Modules/ServerTypes/Activation/ActivationColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ActivationRow } from "./ActivationRow";

export interface ActivationColumns {
    Id: Column<ActivationRow>;
    PlayListTitle: Column<ActivationRow>;
    TeacherPrn: Column<ActivationRow>;
    ActivationLogCode: Column<ActivationRow>;
    DeviceId: Column<ActivationRow>;
    DeviceDetails: Column<ActivationRow>;
    ActivationDate: Column<ActivationRow>;
    ExpiryDate: Column<ActivationRow>;
    EStatus: Column<ActivationRow>;
}

export class ActivationColumns extends ColumnsBase<ActivationRow> {
    static readonly columnsKey = 'Activation.Activation';
    static readonly Fields = fieldsProxy<ActivationColumns>();
}

File: GXpert/GXpert.Web/Modules/Activation/Activation/ActivationColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Activation.Columns;

[ColumnsScript("Activation.Activation")]
[BasedOnRow(typeof(ActivationRow), CheckNames = true)]
public class ActivationColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string PlayListTitle { get; set; }
    public string TeacherPrn { get; set; }

    public string StudentName { get; set; }
    public string ActivationLogCode { get; set; }
    [EditLink]
    public string DeviceId { get; set; }
    public string DeviceDetails { get; set; }
    [QuickFilter]
    public DateTime ActivationDate { get; set; }
    public DateTime ExpiryDate { get; set; }
    [QuickFilter]
    public short EStatus { get; set; }
}